src/Control/Email/Email.php

```diff
@@ -31,738 +16,646 @@ class Email extends ViewableData
+            $this->setSwiftMessage(new Swift_Message());
+        }
+
+        return $this->swiftMessage;
     }
 
     /**
-     * Attach a file based on provided raw data.
+     * @param Swift_Message $swiftMessage
      *
-     * @param string $data The raw file data (not encoded).
-     * @param string $attachedFilename Name of the file that should appear once it's sent as a separate attachment.
-     * @param string|null $mimeType MIME type to use when attaching file. If not provided, will attempt to infer via HTTP::get_mime_type().
      * @return $this
      */
-    public function attachFileFromString($data, $attachedFilename, $mimeType = null)
+    public function setSwiftMessage($swiftMessage)
     {
-        $this->attachments[] = array(
-            'contents' => $data,
-            'filename' => $attachedFilename,
-            'mimetype' => $mimeType,
-        );
+        $swiftMessage->setDate(DBDatetime::now()->Format('U'));
+        if (!$swiftMessage->getFrom() && $defaultFrom = $this->config()->admin_email) {
+            $swiftMessage->setFrom($defaultFrom);
+        }
+        $this->swiftMessage = $swiftMessage;
+
         return $this;
     }
 
     /**
-     * Attach the specified file to this email message.
-     *
-     * @param string $filename Relative or full path to file you wish to attach to this email message.
-     * @param string|null $attachedFilename Name of the file that should appear once it's sent as a separate attachment.
-     * @param string|null $mimeType MIME type to use when attaching file. If not provided, will attempt to infer via HTTP::get_mime_type().
+     * @return string[]
+     */
+    public function getFrom()
+    {
+        return $this->getSwiftMessage()->getFrom();
+    }
+
+    /**
+     * @param string|array $address
+     * @param string|null $name
      * @return $this
      */
-    public function attachFile($filename, $attachedFilename = null, $mimeType = null)
+    public function setFrom($address, $name = null)
     {
-        if (!$attachedFilename) {
-            $attachedFilename = basename($filename);
-        }
-        $absoluteFileName = Director::getAbsFile($filename);
-        if (file_exists($absoluteFileName)) {
-            $this->attachFileFromString(file_get_contents($absoluteFileName), $attachedFilename, $mimeType);
-        } else {
-            user_error("Could not attach '$absoluteFileName' to email. File does not exist.", E_USER_NOTICE);
-        }
+        $this->getSwiftMessage()->setFrom($address, $name);
+
         return $this;
     }
 
     /**
-     * @return string|null
+     * @param string|array $address
+     * @param string|null $name
+     * @return $this
      */
-    public function Subject()
+    public function addFrom($address, $name = null)
     {
-        return $this->subject;
+        $this->getSwiftMessage()->addFrom($address, $name);
+
+        return $this;
     }
 
     /**
-     * @return string|null
+     * @return array
```

Comment:
If I look at SimpleMessage::getSender(), it says `@return string`, so something is wrong here.